internal/config/config.go

```diff
@@ -0,0 +1,162 @@
+package config
+
+import (
+	"fmt"
+	"time"
+
+	ds "github.com/ipfs/go-datastore"
+	dssync "github.com/ipfs/go-datastore/sync"
+	"github.com/ipfs/go-ipns"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/protocol"
+	"github.com/libp2p/go-libp2p-kad-dht/providers"
+	"github.com/libp2p/go-libp2p-kbucket/peerdiversity"
+	record "github.com/libp2p/go-libp2p-record"
+)
+
+// DefaultPrefix is the application specific prefix attached to all DHT protocols by default.
+const DefaultPrefix protocol.ID = "/ipfs"
+
+const defaultBucketSize = 20
+
+// ModeOpt describes what mode the dht should operate in
+type ModeOpt int
+
+// QueryFilterFunc is a filter applied when considering peers to dial when querying
+type QueryFilterFunc func(dht interface{}, ai peer.AddrInfo) bool
+
+// RouteTableFilterFunc is a filter applied when considering connections to keep in
+// the local route table.
+type RouteTableFilterFunc func(dht interface{}, conns []network.Conn) bool
```

Comment:
Note: this is a breaking change as these used to take a `*IpfsDHT`. We could have them take a thing that returns a `host` since that's all that was being used but this might be workable as well while we explore this space.